src/context/CartContext.jsx

```diff
@@ -0,0 +1,143 @@
+import axios from "axios";
+import React, { useState, createContext, useContext, useEffect } from "react";
+import { useAuth } from "./AuthContext";
+import { useToast } from "./ToastContext";
+const CartContext = createContext();
+const initialState = {
+  loading: true,
+  data: [],
+  error: "",
+};
+const CartProvider = ({ children }) => {
+  const { setToast } = useToast();
+  const { token, isLoggedIn } = useAuth();
+  const [cart, setCart] = useState(initialState);
+  const addToCart = (product) => {
+    if (!isLoggedIn) {
+      return;
+    }
+    if (cart.data.find((item) => item._id === product._id)) {
+      setToast({
+        show: true,
+        content: `Item already added to the cart`,
+        type: "info",
+      });
+    } else {
+      axios
```

Comment:
suggestion: If possible try to keep all api calls together

Reply:
Can you tell me why is it necessary to to do so?